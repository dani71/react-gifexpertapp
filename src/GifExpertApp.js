import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

// export function GifExperApp() {

//     return (
//         <>
//             <h2> GifExpertApp</h2>
//             <hr />
//         </>);
// }

export const GifExperApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
    const [categories, setCategories] = useState(['One Punch'])

    // const handleAdd = () => {
    //     // setCategories((c) => [...c,'HunterXHunter']);
    // setCategories([...categories, 'HunterXHunter']);
    // }

    return (
        <div>
            <h2>GifExperApp</h2>
            {/* <AddCategory setCategories={setCategories} /> */}
            <AddCategory setCategories={setCategories} />
            <hr />

            {/* <button onClick={handleAdd}>Aregar</button> */}

            <ol>
                {
                    // categories.map(category => {
                    //     return <li key={category}> {category} </li>
                    // })
                    categories.map(category => (
                        <GifGrid
                            key={ category}
                            category={category}
                        />
                    ))
                    // <li key={category}> {category} </li>)
                }

            </ol>



        </div>
    );
};
