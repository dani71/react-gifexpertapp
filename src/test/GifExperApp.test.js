import React from 'react';
import { shallow } from "enzyme"
import { GifExperApp } from "../GifExpertApp"
// import { GifGrid } from '../components/GifGrid';

describe('Pruebas sobre <GifExperApp />', () => {

    test('Debe coincidir con el snapshot', () => {

        const wrapper = shallow(<GifExperApp />);
        expect(wrapper).toMatchSnapshot();        
    });

    test('debe mostrar una lista de categorÃ­as ', () => {

        const categories = ['One punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExperApp defaultCategories={categories}/>);
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});
