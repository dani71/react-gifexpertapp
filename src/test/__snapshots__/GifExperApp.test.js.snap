// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExperApp /> Debe coincidir con el snapshot 1`] = `
<div>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
